### ISSUE TITLE
Unexpected App Crash

### ISSUE THREAD

**Issue by kalimar** [mapbox/mapbox-gl-native#11473]
**Platform:** Android
**Mapbox SDK version:**  mapbox-android-sdk:5.5.0

### Steps to trigger behavior
  Unknown specifics. Possibly on exiting the activity.

Since March13 we are seeing a crash :
```
Fatal Exception: java.lang.Error
Bad optional Access
com.mapbox.mapboxsdk.storage.FileSource.deactivate (FileSource.java)
com.mapbox.mapboxsdk.maps.MapView.onStop (MapView.java:394)
com.ancestry.ancestrydna.activities.kitResults.ethnicity.view.EthnicityMapActivity.onStop (EthnicityMapActivity.java:693)
```

And the code where it happens :
```java
@Overrideprotected void onStop() { super.onStop();
if (mMap != null) { mMap.onStop(); }
}
```

And if we look deeper into the `mMap.onStop()` we get to `MapView.java`
```
/** * You must call this method from the parent's Activity#onStop() or Fragment#onStop(). */@UiThreadpublic void onStop() { if (mapboxMap != null) { // map was destroyed before it was started mapboxMap.onStop(); } if (mapRenderer != null) { mapRenderer.onStop(); } ConnectivityReceiver.instance(getContext()).deactivate(); FileSource.getInstance(getContext()).deactivate();}
```

And apparently the line ` FileSource.getInstance(getContext()).deactivate();` make the error

**Comment by tobrun** [comment]
I expect that this can occur by closing an activity before it was fully created. We need to check if the optional has a value before accessing it. Adding this to the milestone for a patch release.

**Comment by tobrun** [comment]
fix landed with #11480 and will ship with the upcoming 5.5.1 release.
